src/code/containers/challenge-container-selector.js

```diff
@@ -52,8 +56,10 @@ class ChallengeContainerSelector extends Component {
     if (routeParams.challengeId) {
       routeParams = AuthoringUtils.challengeIdToRouteParams(authoring, routeParams.challengeId);
     }
-    if (hasChangedRouteParams(this.props.currentRouteSpec, routeParams)) {
-      navigateToCurrentRoute({level: routeParams.level-1, mission: routeParams.mission-1, challenge: routeParams.challenge-1});
+    if (isValidRouteParams) {
```

Comment:
Man, good catch on the validity checking...that's where unit tests help. I don't believe your second comment is correct, though: this code path is never run when you enter the navigate URL, because we route directly to the `Navigation` component.